akka-actor/src/main/scala/akka/actor/LightArrayRevolverScheduler.scala

```diff
@@ -128,15 +128,19 @@ class LightArrayRevolverScheduler(config: Config, log: LoggingAdapter, threadFac
         }
       }
 
-      @tailrec final def cancel(): Boolean = {
+      @tailrec private final def tailrecCancel(): Boolean = {
```

Comment:
could this def be defined inside the `def cancel()`?

Reply:
what is the requirement? does it have to be a `private def`? strange  that it wouldn't  allow `final def`

Reply:
The requirement is that the `@tailrec` shouldn't occur on a method overridden from the super-class.
Declared all of them inside the `def` body, Scala 3 compiler accept even that encoding. 👍